src/cli/crd/generate.test.ts

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023-Present The Pepr Authors
+
+import { expect, describe, it, afterEach, jest } from "@jest/globals";
+import {
+  extractSingleLineComment,
+  extractDetails,
+  processSourceFile,
+  uncapitalize,
+  emptySchema,
+  loadVersionFiles,
+  getAPIVersions,
+} from "./generate";
+import { Project } from "ts-morph";
+import * as fs from "fs";
+
+jest.mock("fs");
+
+describe("generate.ts", () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe("extractSingleLineComment", () => {
+    it("should extract a labeled single line comment", () => {
+      const content = `// Kind: Widget`;
+      const result = extractSingleLineComment(content, "Kind");
+      expect(result).toBe("Widget");
+    });
+
+    it("should return undefined if label is missing", () => {
+      const content = `// Group: test`;
+      const result = extractSingleLineComment(content, "Kind");
+      expect(result).toBeUndefined();
+    });
+  });
+
+  describe("uncapitalize", () => {
+    it("should uncapitalize the first letter", () => {
+      expect(uncapitalize("CamelCase")).toBe("camelCase");
+    });
+
+    it("should return empty string when input is empty", () => {
+      expect(uncapitalize("")).toBe("");
+    });
+  });
+
+  describe("emptySchema", () => {
+    it("should return an empty schema object", () => {
+      const result = emptySchema();
+      expect(result).toEqual({ properties: {}, required: [] });
+    });
+  });
+
+  describe("getAPIVersions", () => {
+    it("should return directories from the api root", () => {
+      (fs.readdirSync as jest.Mock).mockReturnValue(["v1", "v2"]);
+      (fs.statSync as jest.Mock).mockImplementation(p => ({
+        isDirectory: () => typeof p === "string" && (p.endsWith("v1") || p.endsWith("v2")),
```

Comment:
```suggestion
      (fs.statSync as jest.Mock).mockImplementation(path => ({
        isDirectory: () => typeof path === "string" && (path.endsWith("v1") || p.endsWith("v2")),
```

I assume `p` represents a filepath on the system,